src/tests/test_magic_cmd.py

```diff
@@ -227,27 +291,62 @@ def test_table_schema_profile(ip, tmp_empty):
     )
 
     expected = {
-        "count": [3],
-        "mean": [22.0],
-        "min": [11.0],
-        "max": [33.0],
-        "std": [11.0],
-        "unique": [3],
-        "freq": [1],
-        "top": [33.0],
+        "count": ["3"],
+        "mean": ["22.0000"],
+        "min": ["11.0"],
+        "max": ["33.0"],
+        "std": ["11.0000"],
+        "unique": ["3"],
+        "freq": [math.nan],
+        "top": [math.nan],
     }
 
     out = ip.run_cell("%sqlcmd profile -t t --schema b_schema").result
 
     stats_table = out._table
 
     for row in stats_table:
-        criteria = row.get_string(fields=[" "], border=False).strip()
+        profile_metric = _get_row_string(row, " ")
 
         cell = row.get_string(fields=["n"], border=False, header=False).strip()
 
-        if criteria in expected:
-            assert cell == str(expected[criteria][0])
+        if profile_metric in expected:
+            assert cell == str(expected[profile_metric][0])
+
+
+def test_table_profile_warnings_styles(ip, tmp_empty):
+    ip.run_cell(
+        """
+    %%sql sqlite://
+    CREATE TABLE numbers (rating float,price varchar(50),number int,word varchar(50));
+    INSERT INTO numbers VALUES (14.44, '2.48', 82, 'a');
+    INSERT INTO numbers VALUES (13.13, '1.50', 93, 'b');
+    """
+    )
+    out = ip.run_cell("%sqlcmd profile -t numbers").result
+    stats_table_html = out._table_html
+    assert "Columns `price` have a datatype mismatch" in stats_table_html
+    assert "#profile-table td:nth-child(3" in stats_table_html
+    assert "Following statistics are not available in" in stats_table_html
+
+
+def test_table_profile_is_numeric(ip, tmp_empty):
```

Comment:
This is still testing the overall functionality and not calling `_is_numeric` directly. What I meant was you can add different test cases like integer, integer as string, non integer (string, null value) etc, basically all possible values that can occur in the dataset and test `_is_numeric` directly

Reply:
Ok, got it!

Reply:
Added the test function `test_profile_is_numeric` that implements the above changes.